server/utils/utils.ts

```diff
@@ -21,3 +23,25 @@ export const initialiseName = (fullName?: string): string | null => {
   const array = fullName.split(' ')
   return `${array[0][0]}. ${array.reverse()[0]}`
 }
+
+/**
+ * Converts input for a GDS date input https://design-system.service.gov.uk/components/date-input/
+ * into a javascript date object
+ * @param dateInputObj an object with date parts (i.e. `-month` `-day` `-year`), which come from a `govukDateInput`.
+ * @param key the key that prefixes each item in the dateInputObj, also the name of the property which the date object will be returned in the return value.
+ * @returns name converted to proper case.
+ */
+export const convertDateInputsToDateObj = <K extends string | number>(dateInputObj: ObjectWithDateParts<K>, key: K) => {
+  const [day, month, year] = [
+    Number(dateInputObj[`${key}-day`]),
+    Number(dateInputObj[`${key}-month`]),
+    Number(dateInputObj[`${key}-year`]),
+  ]
+  const o: { [P in K]?: Date } = dateInputObj
+  if (day && month && year) {
+    o[key] = new Date(Date.UTC(year, month - 1, day, 0, 0, 0))
+  } else {
+    o[key] = new Date(0, 0, 0)
+  }
```

Comment:
I wonder if we should raise an error if there is no date present. We don't really want to pass an invalid date to the API

Reply:
I did think about that... this would result in a strange check in the API but otherwise we would have to show a validation error to the user which might be awkward to implement if all the other validaiton is handled via the API